src/main/java/duke/sonautil/Parser.java

```diff
@@ -0,0 +1,236 @@
+                LocalDate date = LocalDate.parse(dateTime.substring(0, 10));
+                int hour = Integer.parseInt(dateTime.substring(10, 12));
+                int min = Integer.parseInt(dateTime.substring(12));
+                dateTimeComplete = date.atTime(hour, min);
+                timeEntered = "true";
+
+            } else if (dateTime.length() == 10) { //date only
+                LocalDate date = LocalDate.parse(dateTime.substring(0, 10));
+                dateTimeComplete = date.atTime(0, 0);
+
+                //date & time invalid
+            } else {
+                throw new DukeException(Ui.dateTimeErrorMessage());
+            }
+
+            command[0] = "event";
+            command[1] = taskDetails[0];
+            command[2] = dateTimeComplete.toString();
+            command[3] = timeEntered;
+
+        } else if (userMessage.startsWith("delete") && split[0].equals("delete")) {
+
+            if (userMessage.replaceAll(" ", "").equals("delete")) {
+                throw new DukeException(Ui.taskRemoveEmptyMessage());
+            }
+
+            //invalid number
+            boolean isValidCommand = userMessage.toLowerCase().matches("^delete \\d+|^delete -\\d+");
+            if (!isValidCommand) {
+                throw new DukeException(Ui.userUnknownInputMessage(userMessage));
+            }
+
+            String clean = userMessage.substring(6).replaceAll(" ", "");
+
+            //empty message after delete
+            if (clean.equals("")) {
+                throw new DukeException(Ui.taskRemoveEmptyMessage());
+            }
+
+            int taskIndex = Integer.parseInt(clean) - 1;
+
+            command[0] = "delete";
+            command[1] = String.valueOf(taskIndex);
+
+        } else if (userMessage.startsWith("schedule") && split[0].equals("schedule")) {
+
+            String dateString = userMessage.substring(8).replaceAll(" ", "");
+
+            //no date given
+            if (dateString.equals("")) {
+                throw new DukeException(Ui.scheduleNoDateErrorMessage());
+            }
+
+            //date is invalid
+            LocalDate date = LocalDate.parse(dateString);
+            command[0] = "schedule";
+            command[1] = date.toString();
+
+        } else if (userMessage.startsWith("find") && split[0].equals("find")) {
+
+            //only "find"
+            if (userMessage.length() == 4) {
+                throw new DukeException(Ui.findNoKeywordError());
+            }
+
+            String keyword = userMessage.substring(5);
+
+            //no keyword given
+            if (keyword.replaceAll(" ", "").equals("")) {
+                throw new DukeException(Ui.findNoKeywordError());
+            }
+
+            command[0] = "find";
+            command[1] = keyword;
+
+        } else {
+            command[0] = "unknown";
+            throw new DukeException(Ui.userUnknownInputMessage(userMessage));
+        }
+
+        return command;
+    }
```

Comment:
Consider using enums here to make the processing of the user input simpler (e.g. create an Enum for task types and pass the type here for easier processing, rather than repeated calls to "startsWith()". Also, any reason why you did not use a switch statement here rather than repeated using "else if"?